src/angular-modal-service.js

```diff
@@ -180,32 +180,36 @@ module.factory('ModalService', ['$animate', '$document', '$compile', '$controlle
             //  Let angular remove the element and wait for animations to finish.
             $animate.leave(modalElement)
                     .then(function () {
-                      //  Resolve the 'closed' promise.
-                      closedDeferred.resolve(result);
+                      // prevent error if modal is already destroyed
+                      if (modalElement) {
```

Comment:
Suggest you invert the logic here, i.e. bail out if modalElement is falsey, then there's less indentation for the rest of the code!